components/movementsensor/wheeledodometry/wheeledodometry.go

```diff
@@ -156,39 +156,44 @@ func (o *odometry) Reconfigure(ctx context.Context, deps resource.Dependencies,
 	// check if new motors have been added, or the existing motors have been changed, and update the motorPairs accorodingly
 	for i := range newConf.LeftMotors {
 		var motorLeft, motorRight motor.Motor
-		if i >= len(o.motors) || o.motors[i].left.Name().ShortName() != newConf.LeftMotors[i] {
-			motorLeft, err = motor.FromDependencies(deps, newConf.LeftMotors[i])
-			if err != nil {
-				return err
-			}
-			properties, err := motorLeft.Properties(ctx, nil)
-			if err != nil {
-				return err
-			}
-			if !properties.PositionReporting {
-				return motor.NewPropertyUnsupportedError(properties, newConf.LeftMotors[i])
-			}
+		var changed, added bool
```

Comment:
Neither of these are used for another 30 lines. I'd define them around line 184, just before their first use. Up here, they add complexity to the current scope that a human will think they need to remember, even though they don't.

edit: and down below, I think we should inline the checks and get rid of the variables entirely.